spec/requests/posts_controller_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+  let(:user) { create(:user) }
+  let(:post) { create(:post, author: user) }
+
+  describe 'GET #index' do
+    it 'assigns @posts and renders the index template' do
+      get :index, params: { user_id: user.id }
+
+      expect(assigns(:posts)).to eq(user.posts)
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe 'GET #show' do
+    it 'assigns @post and renders the show template' do
+      get :show, params: { id: post.id }
+
+      expect(assigns(:post)).to eq(post)
+      expect(response).to render_template(:show)
+    end
+  end
```

Comment:
- [ ] As required for the project, kindly ensure that for each action you test all the following tests are present(test **If response status was correct**, **If a correct template was rendered**, and **If the response body includes correct placeholder text.)** as you miss some of them. Only one test has checked in this `index` action.